pkg/kubectl/cmd/delete.go

```diff
@@ -223,7 +231,7 @@ func (o *DeleteOptions) Validate(cmd *cobra.Command) error {
 			// To preserve backwards compatibility, but prevent accidental data loss, we convert --grace-period=0
 			// into --grace-period=1 and wait until the object is successfully deleted. Users may provide --force
 			// to bypass this wait.
-			o.WaitForDeletion = true
+			o.WaitForGracefulDeletion = true
```

Comment:
it is very odd to me that --wait does not inform this setting... where I can specify --wait=false, and it still waits